### ISSUE TITLE
SQL Injection possible in cohort definition

### ISSUE THREAD

**Issue by ttemnikova** [OHDSI/circe-be#178]

**Comment by clarkevans** [comment]
This seems like a very high priority ticket.  Is this a systematic problem?

**Comment by chrisknoll** [comment]
I checked our TextFilter criteria types, and we do [this](https://github.com/OHDSI/circe-be/blob/4504aa27efa6bd16ab247b969e740fb005daa6a8/src/main/java/org/ohdsi/circe/cohortdefinition/builders/BuilderUtils.java#L135).

This output was a test I just did using the proposed input.  The `'` were properly escaped into `''`.

```
WHERE C.specimen_source_id  like '%''; DROP TABLE students; SELECT%'
```

Are you still seeing this behavior?

**Comment by clarkevans** [comment]
This was originally reported in Atlas.  Perhaps that report did not use an updated version of Circe, that included https://github.com/OHDSI/circe-be/pull/110.

**Comment by chrisknoll** [comment]
Yes, you would find the test [here](https://github.com/OHDSI/circe-be/blob/e3a4016c21f8a67f7125f44f50ddbf03eb3c5e4e/src/test/java/org/ohdsi/circe/cohortdefinition/builders/BuilderUtilsTest.java#L15).
